src/main/java/com/mindera/school/spaceshiprent/service/rent/RentServiceImpl.java

```diff
@@ -103,7 +129,7 @@ public RentDetailsDto updatePickUpDate(Long id) {
             return converter.convertToRentDetailsDto(rentEntity.get());
         }
 
-        return null;
+        throw new RentNotFoundException(String.format(ErrorMessageConstants.RENT_NOT_FOUND, id));
```

Comment:
you should throw the exception ahead of the rest of the implementation. this way the code is more readable, as the exception is thrown close to the point where you know there is no rent with the given id and the rest of the code will be outside of any if statement.
Look to the `updateRent` method